core/src/banking_stage/packet_receiver.rs

```diff
@@ -8,50 +8,85 @@ use {
         unprocessed_transaction_storage::UnprocessedTransactionStorage,
     },
     crossbeam_channel::RecvTimeoutError,
-    solana_measure::measure::Measure,
-    solana_sdk::{
-        saturating_add_assign,
-        timing::{duration_as_ms, timestamp},
-    },
-    std::{
-        sync::atomic::Ordering,
-        time::{Duration, Instant},
-    },
+    solana_measure::{measure, measure::Measure, measure_us},
+    solana_sdk::{saturating_add_assign, timing::timestamp},
+    std::{sync::atomic::Ordering, time::Duration},
 };
 
 pub struct PacketReceiver;
 
 impl PacketReceiver {
-    #[allow(clippy::too_many_arguments)]
     /// Receive incoming packets, push into unprocessed buffer with packet indexes
     pub fn receive_and_buffer_packets(
         packet_deserializer: &mut PacketDeserializer,
-        recv_start: &mut Instant,
-        recv_timeout: Duration,
         id: u32,
         unprocessed_transaction_storage: &mut UnprocessedTransactionStorage,
         banking_stage_stats: &mut BankingStageStats,
         tracer_packet_stats: &mut TracerPacketStats,
         slot_metrics_tracker: &mut LeaderSlotMetricsTracker,
     ) -> Result<(), RecvTimeoutError> {
-        let mut recv_time = Measure::start("receive_and_buffer_packets_recv");
-        let ReceivePacketResults {
+        let (result, recv_time_us) = measure_us!({
+            let recv_timeout = Self::get_receive_timeout(unprocessed_transaction_storage);
+            let mut recv_and_buffer_measure = Measure::start("recv_and_buffer");
+            let receive_packet_results = packet_deserializer.receive_packets(
+                recv_timeout,
+                unprocessed_transaction_storage.max_receive_size(),
+            )?;
```

Comment:
hmm, i think this `?` is a bit tricky. I think this causes a early bail out, while not calling `.increment_receive_and_buffer_packets_us()` if `.handle_received_packets()` returns `Err`. so, this changes the definition of `slot_metrics_tracker`, which should be avoided...

maybe, quick wrap and run with anonymous closure will fix:

```rust
let (...) = measure_us!(|| -> Result<_, _> {
    ...
}());
```

that said, these are still error-prone.... ;)

Reply:
Yeah you're right that this was still early-exiting 😢 . I suppose I was thinking of measure as a fn taking a closure, but that's not what is does!

I'm not a huge fan of the "naked" immediately executed closures - I went with a `map` instead, which I suppose is doing an immediately called closure internally, but I think makes it more obvious why.